releasenotes/notes/upgrade-curve-fit-4dc01b1db55ee398.yaml

```diff
@@ -0,0 +1,46 @@
+---
+upgrade:
+  - |
+    The :class:`CurveAnalysis` class has been updated to use the covariance between fit
+    parameters in the error propagation. This will provide more accurate standard
+    error for your fit values.
+  - |
+    The data format of analysis result data value has been replaced from
+    :class:`FitVal` to :class:`UFloat` to support error propatation for
+    post analysis computation.
+
+    .. code-block:: python
+
+      expdata = T1(0, delays, backend).run()
+      fit_t1 = expdata.analysis_results("T1").value
+
+      assert isinstance(fit_t1, UFloat)
+
+      new_value = fit_t1 / 2
+      new_value.std_dev  # show new standard error value
+
+    Now ``fit_t1`` value is :class:`UFloat` instance with
+    new properties :attr:`.nominal_value` and :attr:`.std_dev`,
+    and you can directly apply methematical operation to this object.
+    New error value is predicted by linear error propagation theory.
+    Note that you no longer need to separately compute standard error.
+
+    This computation is offered by the uncertainties (https://pythonhosted.org/uncertainties/)
+    package which is a requirement since from Qiskit Experiments v0.2.
+    The functions supporting error propagation is also available in ``uncertainties.umath``.
+  - |
+    The curve fit plot generated by the :class:`CurveAnalysis` shows a
```

Comment:
This one is more a new feature than an upgrade: Something like "added a <new option name> option to curve analysis to allow plotting custom confidence intervals during analysis. The default option values plot the 1 and 3 sigma cofidence intervals.

Aside: I don't really get why do we have the 3 sigma as the default? Why not just the 1-sigma? or 1 and 2 sigma? Or 1,2,3 ? Seems odd to me to skip 2 sigma and go straight to 3.

Reply:
Make sense. 3 comes from empirical statistics (three-sigma rule) https://www.oxfordreference.com/view/10.1093/oi/authority.20110803104447825